include/deal.II/base/config.h.in

```diff
@@ -192,6 +193,14 @@
     >=  \
     (major)*10000 + (minor)*100 + (subminor))
 
+
+/*
+ * gmsh:
+ */
+#ifdef DEAL_II_WITH_GMSH
+#  define GMSH_EXECUTABLE @GMSH_EXECUTABLE@
```

Comment:
Maybe call it `DEAL_II_GMSH_EXECUTABLE` to not tread on other packages' namespaces?

Reply:
This is consistent with all other packages. We don't set `DEAL_II_PETSC_INCLUDE_DIRECTORIES`, but `PETSC_INCLUDE_DIRECTORIES`. This is a requirement for any `FindPACKAGE.cmake` in `cmake` standards, as far as I know, and this is how we do it for all other packages in `cmake/modules/FindXXX.cmake`.
This is done through `cmake/macros/macro_deal_ii_package_handle.cmake`.

Reply:
Oh, I see, maybe you mean the define inside the code, not inside cmake variables... :P I'll do that.

Reply:
Yes, correct. The name of the `@...@` expanded variable doesn't matter -- it never escapes cmake. But the `#define`d variable can conflict with other packages.